pkg/R/sparkRBackend.R

```diff
@@ -72,18 +72,21 @@ invokeJava <- function(isStatic, objId, methodName, ...) {
   writeInt(rc, length(args))
   writeArgs(rc, args)
 
+  # Construct the whole request message to send it once,
+  # avoiding write-write-read pattern in case of Nagle's algorithm
```

Comment:
Would be good to add a reference to [this post by Nagle himself](http://developers.slashdot.org/comments.pl?sid=174457&cid=14515105) as well as SPARKR-183 to explain what's going on.